### DISCUSSION TITLE
maximize data retention

### DISCUSSION THREAD

**Discussion by xandovia** [Security-Onion-Solutions/securityonion#8709] [Unsupported Versions]
According to the "EU Data Retention Directive" policy, I need to store the users' connection data (maybe connection logs or src/dest IP/port only or whatever fits best).
I think SecurityOnoin is a powerful option, but the point is that it needs lots of storage. In my standalone test installation, even with a disabled Stenographer, there is still an 80GB Elasticsearch indices file within a week.

- Our average download and upload speeds are about 80Mbps most of the time
- At least one remote site (planning to install a sensor there)
- A maximum of 350 users and perhaps 500 active devices are expected.
- According to ES logs, 12 million connections are made each day.

I would like to know:
1. Is there a better way to keep user connection logs?
2. Is it possible to disable more services to reduce the size of the logs and indices?

Any suggestions would be greatly appreciated.

**Comment by dougburks** [comment]
One option might be to keep your short-term searchable data in Elastic indices but have your long-term retention in the form of the raw Zeek connection logs. You can find them in date stamped directories in `/nsm/zeek/logs/` and they are already gzipped to save space.
